lib/kubeclient.rb

```diff
@@ -345,7 +345,8 @@ def create_faraday_client
           cert_store: @ssl_options[:cert_store],
           client_cert: @ssl_options[:client_cert],
           client_key: @ssl_options[:client_key],
-          verify: @ssl_options[:verify_ssl]
+          verify: @ssl_options[:verify_ssl] != OpenSSL::SSL::VERIFY_NONE,
+          verify_mode: @ssl_options[:verify_ssl]
```

Comment:
Oh wow good find :clap:

This should fix the issue as long as users pass integers 0=VERIFY_NONE or 1=VERIFY_PEER, which has been the convention kubeclient expects so far :heavy_check_mark:  

But this confusion makes me wonder what happens if a user of kubeclient gets confused about what _they_ should pass.  E.g. if they pass a bool, both false and true will result in `verify: true` because true != 0, and `verify_mode` will pass an unexpected bool into faraday (whose effect may depend on adapter)...

So I'm thinking should we also add validation in the constructor that `ssl_options[:verify_ssl]` is one of VERIFY_NONE/VERIFY_PEER intergers?
WDYT? (but not in this PR, I may want to backport that one validation)

Reply:
that is a good question, further down in the file there is another use of the same verify_ssl value, i think that the most sensible thing would be to validate that it's `OpenSSL::SSL::VERIFY_NONE || OpenSSL::SSL::VERIFY_PEER` we could try and convert `true/false` but maybe it's just better to throw an error and to instead require the correct SSL values, because they could change 🤷 

other usage: https://github.com/ManageIQ/kubeclient/blob/master/lib/kubeclient.rb#L726